core/sdk-core/src/main/java/software/amazon/awssdk/core/useragent/BusinessMetrics.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.useragent;
+
+import static software.amazon.awssdk.core.internal.useragent.UserAgentConstant.COMMA;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.UnaryOperator;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.utils.CollectionUtils;
+
+// CHECKSTYLE:OFF - Class naming rules prohibit plurals, overriding
+@SdkProtectedApi
+public class BusinessMetrics {
```

Comment:
nit: final. Also are we deadset on the name, since we're ignoring checkstyle for it? If not, I'd vote for just using a name that follows our conventions.

Reply:
Discussed offline and settled on BusinessMetricsCollection